### PR TITLE
feat: export and import comments as description for Android XML

### LINKED ISSUE #2391: export description with strings as comment for Android
**Is your feature request related to a problem? Please describe.**
When exporting strings.xml for android the description in tolgee is missing.

**Describe the solution you'd like**
before every string that contains a not empty description there should be a comment 
`<!-- this is a comment in xml -->`

### PR DESCRIPTION
**Pull request by Anty0**
Supersedes  #2472
Fixes  #2391 and  #2464

### CODE REVIEW

File: backend/data/src/main/kotlin/io/tolgee/formats/android/out/AndroidStringsXmlFileWriter.kt [hunk 1]
```diff
@@ -67,11 +66,13 @@ class AndroidStringsXmlFileWriter(private val model: AndroidStringsXmlModel) {
...
-        this.appendChild(imported)
-      }
+    contentToAppend.children?.forEach {
+      val imported = this.ownerDocument.importNode(it, true)
+      this.appendChild(imported)
     }
   }
+
+  private fun Element.optionalComment(comment: String?) {
+    comment?.let { comment(" $it ") }
```
Comment (Anty0) [inline backend/data/src/main/kotlin/io/tolgee/formats/android/out/AndroidStringsXmlFileWriter.kt:76]: I prefer how the comments look when there are spaces around them. Is it ok to do this?

Comment (JanCizmar) [reply backend/data/src/main/kotlin/io/tolgee/formats/android/out/AndroidStringsXmlFileWriter.kt:76]: Yes

Comment (JanCizmar) [inline backend/data/src/main/kotlin/io/tolgee/formats/android/out/AndroidStringsXmlFileWriter.kt:76]: Don't add it if it's empty.

**Review by JanCizmar — CHANGES_REQUESTED** [review]

**Review by JanCizmar — APPROVED** [review]

File: backend/data/src/main/kotlin/io/tolgee/formats/android/out/AndroidStringsXmlFileWriter.kt (hunk 1 above)
Comment (Anty0) [reply backend/data/src/main/kotlin/io/tolgee/formats/android/out/AndroidStringsXmlFileWriter.kt:76]: good point

**Pull request merged**